AG-10875 Fix navigator bounding box region

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-10875

* When the min/max handles are on the edge of masks they stick out a bit, so take this offset into account. This fixes a bug where the cursor begin reset because the cursor left the region, and set back to 'ew-cursor' immediately after. (copy of https://github.com/ag-grid/ag-charts/pull/1130)
* Reset cursor in the navigator hover handler.
* Lock the `RegionManager.currentRegion` when dragging. This simplifies the navigator input handling logic.

olegat committed:
AG-10875 Fix navigator bounding box region

When the min/max handles are on the edge of masks they stick out a bit,
so take this offset into account. This fixes a bug where the cursor
begin reset because the cursor left the region, and set back to
'ew-cursor' immediately after.

olegat committed:
AG-10875 Reset cursor in navigator hover handler

olegat committed:
AG-10875 Lock the RegionManager.currentRegion when dragging

This change stops the RegionManager from updating the currentRegion when
the drag is currently happening. A hover or leave event is then sent
alongside a drag-end event when the mouse button is released.

This simplies the input handling logic of the navigator.

lsjroberts commented on packages/ag-charts-community/src/chart/navigator/shapes/rangeSelector.ts:
```diff
@@ -44,6 +50,8 @@ export class RangeSelector extends Group {
-        return new BBox(x, y, width, height);
+        const minOff = this.minHandle.width / 2;
+        const maxOff = this.maxHandle.width / 2;
+        return new BBox(x - minOff, y, width + (minOff + maxOff), height);
```
The goal in my refactor was to make these scenes internally independent so it was more explicit in the `Navigator` class what was happening, as hiding this stuff in the scenes was causing lots of issues with the zoom.

So I'd prefer if in `Navigator.layoutNodes()` it passed in something like:

```
private layoutNodes(x: number, y: number, width: number, height: number) {
    // ...
    rangeSelector.layout(x - minOffset, y, width + (minOffset - maxOffset), height);
    // ...
}
```

So the `RangeSelector` doesn't need to know about the handles and mask directly, and instead the scenes are coordinated by the parent `Navigator`.

olegat replied:
Done. Moved the handle offset calculations to `Navigator.layoutNodes()`

olegat committed:
AG-10875 CR: Specify layout of RangeSelector in parent Navigator class

olegat committed:
AG-10875 Fix regression tests

Changing the {x,y,width} values of RangeSelector causes the minichart to
be drawn a little bit to the left instead of in the middle. So instead,
use the left/right offset only the computation of RangeSelector BBox.

lsjroberts reviewed (approved):
Nice, lgtm
